script.js

```diff
@@ -1,4 +1,142 @@
+// ------ ROCK PAPER SCISSORS -----//
+var userPoints = 0;
+var comPoints = 0;
+var userName = '';
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+// get userName as user's first input
+if (!userName){
+  if (!input){
+  return 'Please input username';
+}
+  userName = input;
+  return 'Hello ' +userName+ ' ! To start playing, please enter "scissors", "paper" or "rock"!'
+}
+if (input == 'scissors' || input == 'paper' || input == 'rock'){
+  var comPlay = convertNumbertoComPlay(randomNumber());
+  return gamePlay(input,comPlay);
+}
 };
+
+
+var randomNumber = function(){
+  var randomDecimal = Math.random()*3;
+  var randomInteger = Math.floor(randomDecimal);
+  return randomInteger; 
+};
+
+var convertNumbertoComPlay = function(number){
+  if (number == 1){
+    return 'scissors';
+  }
+  
+  else if (number == 2){
+    return 'paper';
+  }
+  
+  else if (number == 0){
+    return "rock";
+  }
+};
+
+var gamePlay = function(userPlay, comPlay){
+  if (userPlay == comPlay){
+    return 'It is a tie! <br> Both you and computer played ' +userPlay+ '. The current record is: user: ' +userPoints+ ' computer: '+comPoints ;
+  }
+
+  else if (userPlay == 'scissors'){
+    if (comPlay == 'paper'){
+      userPoints = userPoints + 1;
+      return 'Yay! You won! <br> Computer played ' +comPlay+ ' and you played ' +userPlay+ '. <br> The current record is: user: ' +userPoints+ ' computer: '+comPoints ;
+    }
+    else { 
+      comPoints = comPoints +1;
+      return 'Aww...you lost! <br> Computer played ' +comPlay+ ' and you played ' +userPlay+ '. <br> The current record is: user: ' +userPoints+ ' computer: '+comPoints;
+    }
+  }
+
+  else if (userPlay == 'paper'){
+    if (comPlay == 'rock'){
+      userPoints = userPoints + 1;
+      return 'Yay! You won! <br> Computer played ' +comPlay+ ' and you played ' +userPlay+ '. <br> The current record is: user: ' +userPoints+ ' computer: '+comPoints ;
+  }
+  else {
+    comPoints = comPoints +1;
+    return 'Aww...you lost! <br> Computer played ' +comPlay+ ' and you played ' +userPlay+ '. <br> The current record is: user: ' +userPoints+ ' computer: '+comPoints ;
+  }
+}
+  else if (userPlay == 'rock'){
+    if (comPlay == 'scissors'){
+      userPoints = userPoints + 1;
+      return 'Yay! You won! <br> Computer played ' +comPlay+ ' and you played ' +userPlay+ '. <br> The current record is: user: ' +userPoints+ ' computer: '+comPoints ;
+    }
+    else{
+      comPoints = comPoints +1;
+      return 'Aww...you lost! <br> Computer played ' +comPlay+ ' and you played ' +userPlay+ '. <br> The current record is: user: ' +userPoints+ ' computer: '+comPoints ;
+    }
+  }
+  else{
+    return 'Invalid response! <br> Please input only: scissors, paper or rock.'
```

Comment:
This line is redundant because in line 14 ,  if (input == 'scissors' || input == 'paper' || input == 'rock') meaning userPlay will be  either scissors ,paper or rock.


else{
    return 'Invalid response! <br> Please input only: scissors, paper or rock.'
} 
should be in line 18